### PR TITLE
feat: support new erc20

### PR DESCRIPTION
**Pull request by janek26**
This PR implements the new ERC20 as defined in openzeppelins repo, with shortstring symbol and name aswell as uint256 for token balances.

(!) The contract abi parsing needs to be revisited at some point to make it easier

### CODE REVIEW

File: packages/extension/src/assets/erc20-tokens.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [
   {
-    "address": "0x00a45e3942b7a75983dea7afffda9304d0273773619d1e3d5eaa757d751bfaf3",
+    "address": "0x07394cbe418daa16e42b87ba67372d4ab4a5df0b05c6e554d158458ce245bc10",
```
Comment (delaaxe) [inline packages/extension/src/assets/erc20-tokens.json:3]: On which network is this? Network should be in an extra field here IMO

File: packages/extension/src/ui/App.tsx [hunk 2]
```diff
@@ -21,6 +23,13 @@ import { WelcomeScreen } from "./screens/WelcomeScreen"
 import { routerMachine } from "./states/RouterMachine"
 import { TokenDetails } from "./utils/tokens"
 
+function getUint256CalldataFromBN(bn: BigNumber) {
+  return {
+    type: "struct" as "struct",
```
Comment (delaaxe) [inline packages/extension/src/ui/App.tsx:28]:
```suggestion
    type: "struct" as const,
```

File: packages/extension/src/ui/utils/tokens.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ import { isValidAddress } from "./addresses"
 export const playgroundToken = erc20Tokens.find(
   ({ address }) =>
     address ===
-    "0x00a45e3942b7a75983dea7afffda9304d0273773619d1e3d5eaa757d751bfaf3",
+    "0x07394cbe418daa16e42b87ba67372d4ab4a5df0b05c6e554d158458ce245bc10",
```
Comment (delaaxe) [inline packages/extension/src/ui/utils/tokens.ts:13]: This should depend on network

File: packages/playground/src/services/token.service.ts [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
 export const erc20TokenAddress =
-  "0x00a45e3942b7a75983dea7afffda9304d0273773619d1e3d5eaa757d751bfaf3"
+  "0x07394cbe418daa16e42b87ba67372d4ab4a5df0b05c6e554d158458ce245bc10"
 
 const mintSelector = stark.getSelectorFromName("mint")
 
 const transferSelector = stark.getSelectorFromName("transfer")
 
+function getUint256CalldataFromBN(bn: number.BigNumberish) {
+  return { type: "struct" as "struct", ...uint256.bnToUint256(bn) }
```
Comment (delaaxe) [inline packages/playground/src/services/token.service.ts:13]:
```suggestion
  return { type: "struct" as const, ...uint256.bnToUint256(bn) }
```

**Review by delaaxe — APPROVED** [review]

**Comment by janek26** [comment]
We should bind the token to a specific network as soon as Maine supports its first ERC20

**Pull request merged**